### PR TITLE
Regexp: Responsive table design

### PR DESCRIPTION
**Pull request by NaveenKarippai**
###### What does your Pull Request do (check all that apply)?

Choose the most relevant items and use the following title template to name
your Pull Request.
- [ ] New Instant Answer
  - [ ] Cheat Sheets: **`New {Cheat Sheet Name} Cheat Sheet`**
  - [ ] Other: **`New {IA Name} Instant Answer`**
- [x] Improvement
  - [ ] Bug fix: **`{IA Name}: Fix {Issue number or one-line description}`**
  - [x] Enhancement: **`regexp: {2643} make table output responsive in design`**
- [ ] Non–Instant Answer
  - [ ] Other (Role, Template, Test, Documentation, etc.): **`{GoodieRole/Templates/Tests/Docs}: {Short Description}`**
###### Description of changes

The goodie output table was not responsive on small screen
devices like tablet on mobile phones. This could be fixed by
some CSS styling magic.

Provide an overview of the changes this pull request introduces.
- CSS media was used to detect screen width
- responsive on small screen < 760px
###### Which issues (if any) does this fix?

See also: #2643

Fixes #NNNN - how specifically does it fix the issue?
###### People to notify (@GuiltyDolphin, @moollaza, @tagawa)

---

Instant Answer Page: https://duck.co/ia/view/regexp

[Maintainer](http://docs.duckduckhack.com/maintaining/guidelines.html): @jagtalon

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: share/goodie/regexp/regexp.css [hunk 1]
```diff
@@ -2,3 +2,17 @@
...
 }
+
+/* responsive table */
+@media
+only screen and (max-width: 760px),
+(min-device-width: 768px) and (max-device-width: 1024px)  {
+.zci--regexp .record .record__cell {
+        display: block;
+        border-top: 1px solid #d0d0d0;
+	border-bottom: 1px solid #eee;
```
Comment (moollaza) [inline share/goodie/regexp/regexp.css]: nit: indentation isn't consistent

**Comment by moollaza** [comment]
Deploying to beta...

**Comment by moollaza** [comment]
@NaveenKarippai it seems like this fix may be a little overkill -- We seem to have the width of the first `<td>` set to 20em, ignoring the width of the page.

It may be better to set the width to a max of 35-50%, and right align the 2nd `<td>`

We can also save a lot of horizontal space by rewording `Subpattern Match` to `Group`

**Comment by GuiltyDolphin** [comment]
@moollaza This is the main case I am concerned with https://beta.duckduckgo.com/?q=regexp+%2F%28.*%29%28%3F%3Cfooooooooooooooooooooooooooooo%3E.%29%2F+ddg&ia=answer

(And pretty much everything on mobile).

I'd rather the name stayed as `Subpattern Match` on desktop - though maybe `Group` would be more suitable on mobile (feel free to veto).

I initially hard-coded the width as it seemed a pretty sensible amount, providing a fairly nice display for most sensible-length names; unfortunately it does not do anything good for the eyes on mobile.

**Comment by moollaza** [comment]
> This is the main case I am concerned with https://beta.duckduckgo.com/?q=regexp+%2F%28.*%29%28%3F%3Cfooooooooooooooooooooooooooooo%3E.%29%2F+ddg&ia=answer

I understand where you're going, but I think that's a _bit_ of an edge case? Obviously at some point we have to clip it, but we've got room to expand the desktop UI.

> I'd rather the name stayed as Subpattern Match on desktop - though maybe Group would be more suitable on mobile (feel free to veto).

The reason I suggested `Group` is because most of the Regex testing tools I've used seem to also use that terminology. "Capture group" is more descriptive, but again lengthy. Either way, the terminology should be consistent across designs though.

I can have the design team take a look as well?

**Comment by GuiltyDolphin** [comment]
@moollaza Yeah, some expert opinion wouldn't go amiss :)

**Comment by moollaza** [comment]
@NaveenKarippai @GuiltyDolphin -- design team is a little busy, but there's a simple fix to the immediate problem which is that we have a fixed width of `20em` on mobile. If we remove that content that _should_ fit in the space provided, _will_ fit.

Let's just remove that fixed width for now and then figure out how to handle edge cases where the text is wider than the screen.

**Comment by moollaza** [comment]
regexp________ddg_at_duckduckgo

@NaveenKarippai you can use a rule like:

```css
.is-mobile .zci--regexp .record__cell--key {
    width: auto;
}
```

**Comment by NaveenKarippai** [comment]
@moollaza Could you please review the new changset.

Should we keep the css `media` query?

**Comment by moollaza** [comment]
> Should we keep the css media query?

Yes please. The table _should_ be providing a "show more" button whenever the content is being clipped to allow it to expand which should alleviate the need for the media query

**Comment by NaveenKarippai** [comment]
@moollaza understood. In that case, the current changes  should be sufficient. Could you please make an update if it's ready to merge. Thanks!

**Comment by moollaza** [comment]
@NaveenKarippai whoops just realized I misread your message your question above -- thought you asked if we should **_remove_** the media query -- which we should After that we can merge this.

The search team will need to update the table to make sure we show a "Show more" when necessary

File: share/goodie/regexp/regexp.css [hunk 2]
```diff
@@ -1,4 +1,18 @@
 .zci--regexp .record .record__cell--key {
-    width: 20em;
+    width: auto;
```
Comment (moollaza) [inline share/goodie/regexp/regexp.css]: I think the default is `auto`? We can just remove this rule and it should have the same effect I think?

**Comment by NaveenKarippai** [comment]
@moollaza hm, we have removed unnecessary style properties as you pointed out. I'm not sure about how to proceed further by removing media queries. If there is a similar IA to take inspiration from, it would be awesome!

File: share/goodie/regexp/regexp.css [hunk 3]
```diff
@@ -1,4 +1,17 @@
...
+@media
+only screen and (max-width: 760px),
+(min-device-width: 768px) and (max-device-width: 1024px)  {
+.zci--regexp .record .record__cell {
+        display: block;
+        border-top: 1px solid #d0d0d0;
+	border-bottom: 1px solid #eee;
+	position: relative;
+	padding-right: 60%;
+        overflow:visible;
```
Comment (GuiltyDolphin) [inline share/goodie/regexp/regexp.css]: Nd here.

**Comment by GuiltyDolphin** [comment]
@moollaza @NaveenKarippai This looks good on full size:

screenshot-2016-06-18-19 17 00_745_281

And 'default' view for mobile:

screenshot-2016-06-18-19 17 26_369_514

But if you click 'Show More', it screws up the names:

screenshot-2016-06-18-19 17 36_363_506

I think we should make sure that name stays on the same line regardless of whether they pressed 'Show More' or not.

File: share/goodie/regexp/regexp.css (hunk 3 above)
Comment (NaveenKarippai) [inline share/goodie/regexp/regexp.css]: done!

**Comment by moollaza** [comment]
@NaveenKarippai all we need to do is remove the media query. On mobile, there should be a "Show More" button when the content is clipped. Clicking it will expand the table vertically and the column text will wrap.

On desktop we're not always doing that when we should though and that needs to be fixed in the template. I think the JS code is internal though.

**Comment by NaveenKarippai** [comment]
@moollaza Thanks for the update!

> I think the JS code is internal though

Do I have an access to internal JS code to make the changes?

**Comment by moollaza** [comment]
> Do I have access on internal JS code to make the changes?

Unfortunately not. By internal I mean closed source. We're still working to open source more of the code, but we haven't gotten around to some parts of the template code yet.

Let's get this finished and merged and I can get the team to adjust the internal code.

**Comment by mintsoft** [comment]
Fixed by : https://github.com/duckduckgo/zeroclickinfo-goodies/issues/2643 https://github.com/duckduckgo/zeroclickinfo-goodies/pull/3586

**Pull request closed without merging**